tsl/test/sql/hypertable_generalization.sql

```diff
@@ -0,0 +1,241 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+
+CREATE OR REPLACE FUNCTION part_func(id TEXT)
+	RETURNS INTEGER LANGUAGE PLPGSQL IMMUTABLE AS
+$BODY$
+DECLARE
+	retval INTEGER;
+BEGIN
+	retval := CAST(id AS INTEGER);
+	RETURN retval;
+END
+$BODY$;
+
+SELECT by_range('id');
+SELECT by_range('id', partition_func => 'part_func');
+SELECT by_range('id', '1 week'::interval);
+SELECT by_range('id', '1 week'::interval, 'part_func'::regproc);
+SELECT by_hash('id', 3);
+SELECT by_hash('id', 3, partition_func => 'part_func');
+
+\set ON_ERROR_STOP 0
+SELECT 'hash//id//3//-'::_timescaledb_internal.dimension_info;
```

Comment:
It seems we are only testing that the construction with an invalid string does not work. Could you add a test case to show that the construction with a valid string works?

Reply:
Also note that the string above is "valid" in the sense that it is what the output function outputs.